### ISSUE TITLE
Driver Automation Tool 6.3.0 - Driver package download - Version

### ISSUE THREAD

**Issue by EvilerBetty** [SCConfigMgr/ConfigMgr#172]
It looks as though Dell is listing some of the driver pack releases as "Version 1.0, A13".
Example:  http://downloads.dell.com/published/pages/latitude-12-5290-2-in-1-laptop.html

This results in the Driver Package version being set as "1" when imported into SCCM.

**Comment by vartaxe** [comment]
Oh hell dell is starting to make me crazy

**Comment by Alslinet** [comment]
Biggest problem with this is the driver pack wont update. Since the Version number never changes. Workaround could be as easy as changing Version number to 1.

**Comment by Alslinet** [comment]
The problem is this bit of code: (($DriverCab).Split("-")[2]).Trim(".cab") 
2-in-1 cab files have additional dashes (-) so it kinda screws With the versioning.

Example:
(("FOLDER05672009M/1/5290 2-IN-1-Win10-A09-8FY5R.CAB").split("-")[2]).Trim(".cab")
(("FOLDER05822961M/1/WinPE10.0-Drivers-A15-FYCJR.CAB").split("-")[2]).Trim(".cab")
(("FOLDER05323227M/1/WinPE3.0-Drivers-A23-MJ53D.CAB").split("-")[2]).Trim(".cab")

Results:
1
A15
A23

**Comment by Alslinet** [comment]
Solution is probably to grab the Dell Version number from the xml and not the cab file name.

**Comment by maocul** [comment]
+1 on this issue. my change is probably too simplistic though!

`IF ( $DriverCab -like "*2-in-1* ) { $DriverRevision = (($DriverCab_.Split("-")[4]).Trim(".cab") }
Else { $DriverRevision = (($DriverCab_.Split("-")[2]).Trim(".cab") } `

We're only dealing with Dell hardware and Windows 10 so I've probably broken a bunch of other functions!

**Comment by skywayskase** [comment]
I like the solution suggested by @Alslinet, but unless that's the only model you have, it'll think that every Dell model has that same version. 
If you instead replace `$DriverRevision = (($DriverCab).Split("-")[2]).Trim(".cab")` with something like ```$DriverRevision = ($Global:DellModelCabFiles | Where-Object {(($_.path).Split("/") | Select-Object -Last 1) -match $DriverCab}).dellVersion``` it should scale better.

**Comment by Alslinet** [comment]
@skywayskase, you are right, i forgot to add that. This is what im gonna try when i get back in to work. There is a line added at the end of the if and else sections. 
`if ($global:SkuValue.Count -gt 1) {
	$DellSingleSKU = $global:SkuValue | Select-Object -First 1
	$global:SkuValue = [string]($global:SkuValue -join ";")
	global:Write-LogEntry -Value "Info: Using SKU : $DellSingleSKU" -Severity 1
	$ModelURL = $DellDownloadBase + "/" + ($global:DellModelCabFiles | Where-Object {
			((($_.SupportedOperatingSystems).OperatingSystem).osCode -match $WindowsVersion) -and ($_.SupportedSystems.Brand.Model.SystemID -match $DellSingleSKU)
		}).delta
	$DriverDownload = $DellDownloadBase + "/" + ($global:DellModelCabFiles | Where-Object {
			((($_.SupportedOperatingSystems).OperatingSystem).osCode -match $WindowsVersion) -and ($_.SupportedSystems.Brand.Model.SystemID -match $DellSingleSKU)
		}).path
	$DriverCab = (($global:DellModelCabFiles | Where-Object {
				((($_.SupportedOperatingSystems).OperatingSystem).osCode -match $WindowsVersion) -and ($_.SupportedSystems.Brand.Model.SystemID -match $DellSingleSKU)
			}).path).Split("/") | Select-Object -Last 1
	$DriverRevision = ($global:DellModelCabFiles | Where-Object { ((($_.SupportedOperatingSystems).OperatingSystem).osCode -match $WindowsVersion) -and ($_.SupportedSystems.Brand.Model.SystemID -match $DellSingleSKU) } | Select-Object -Last 1).dellVersion #THIS IS ADDED
}
else {
	$ModelURL = $DellDownloadBase + "/" + ($global:DellModelCabFiles | Where-Object {
			((($_.SupportedOperatingSystems).OperatingSystem).osCode -match $WindowsVersion) -and ($_.SupportedSystems.Brand.Model.SystemID -match $global:SkuValue)
		}).delta
	$DriverDownload = $DellDownloadBase + "/" + ($global:DellModelCabFiles | Where-Object {
			((($_.SupportedOperatingSystems).OperatingSystem).osCode -match $WindowsVersion) -and ($_.SupportedSystems.Brand.Model.SystemID -match $global:SkuValue)
		}).path
	$DriverCab = (($global:DellModelCabFiles | Where-Object {
				((($_.SupportedOperatingSystems).OperatingSystem).osCode -match $WindowsVersion) -and ($_.SupportedSystems.Brand.Model.SystemID -match $global:SkuValue)
			}).path).Split("/") | Select-Object -Last 1
	$DriverRevision = ($global:DellModelCabFiles | Where-Object { ((($_.SupportedOperatingSystems).OperatingSystem).osCode -match $WindowsVersion) -and ($_.SupportedSystems.Brand.Model.SystemID -match $global:SkuValue) } | Select-Object -Last 1).dellVersion #THIS IS ADDED
}
$ModelURL = $ModelURL.Replace("\", "/")
#$DriverRevision = (($DriverCab).Split("-")[2]).Trim(".cab") THIS IS REMOVED`
